server/views/visit.js

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var config = require('../config');
+var url2png = require('url2png')(config.get('url2png_apiKey'), config.get('url2png_secretKey'));
+
+var transform = function(results) {
+  // keep visit properties top level, move all other page properties under a userPage key
+  var newResult = {};
+  ['id', 'url', 'title', 'visited_at'].forEach(function(item) {
+    newResult[item] = results[item];
+    delete results[item];
+  });
+
+  // remove the userId from the visit, if present
+  delete results.userId;
+
+  newResult.userPage = results;
+  return newResult;
+};
+
+var addScreenshot = function(item) {
+  item.screenshot_url = url2png.buildURL(item.url, {viewport: '1024x683', thumbnail_max_width: 540});
```

Comment:
lol, yeah, I think I'll let it look obviously wrong until we start proxying those instead of doing the super-slo-mo API request on the client.